Backend/django_mobx_test/restaurant/views.py

```diff
@@ -0,0 +1,182 @@
+            tickets = Ticket.objects.filter(restaurant=restaurant)
+            for ticket in tickets:
+                ticket.delete()
+
+            restaurant.delete()
+            return Response({'data': status.HTTP_200_OK}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    # Tickets
+    @action(detail=True, methods=['post'])
+    def ticket(self, request, pk=None):
+        try:
+            serializer = TicketSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            restaurant = Restaurant.objects.get(id=pk,  owner=request.user)
+            ticket = Ticket(name=serializer.validated_data['name'], amount=serializer.validated_data['amount'],
+                            max_purchase=serializer.validated_data['max_purchase'], restaurant=restaurant)
+            ticket.save()
+            ticket_data = TicketModelSerializer(ticket)
+            return Response({'data': ticket_data.data}, status=status.HTTP_201_CREATED)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @action(detail=True, methods=['get'])
+    def tickets(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            tickets = Ticket.objects.filter(restaurant=restaurant)
+            ticket_data = TicketModelSerializer(tickets, many=True)
+            return Response({'data': ticket_data.data}, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @action(detail=True, methods=['delete'])
+    def ticket_id(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            tickets = Ticket.objects.get(id=request.query_params['id'], restaurant=restaurant)
+            tickets.delete()
+            return Response({'data': status.HTTP_200_OK}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @action(detail=True, methods=['get'])
+    def specific_ticket(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            tickets = Ticket.objects.get(id=request.query_params['id'], restaurant=restaurant)
+            ticket_data = TicketModelSerializer(tickets)
+            return Response({'data': ticket_data.data}, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class PurchasedTicketViewSet(ViewSet):
+    permission_classes = (AllowAny, )
+
+    def list(self, request):
+        try:
+            tickets = Ticket.objects.all()
```

Comment:
According to requirements, it should show only specific restaurant's tickets, also you can use (or merge) with same viewset of Ticket, but need to just change few things like change endpoint, but it will work on same Ticket model, so would be better covered in same class